smithy-swift-codegen/src/main/kotlin/software/amazon/smithy/swift/codegen/customtraits/NeedsReaderTrait.kt

```diff
@@ -0,0 +1,17 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+
+package software.amazon.smithy.swift.codegen.customtraits
+
+import software.amazon.smithy.model.node.Node
+import software.amazon.smithy.model.shapes.ShapeId
+import software.amazon.smithy.model.traits.Trait
+
+class NeedsReaderTrait : Trait {
```

Comment:
A simple custom trait we can use to mark the struct or union shapes that need to have a reader code-generated for them.